iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/utils/MultiTsFileDeviceIterator.java

```diff
@@ -426,29 +424,27 @@ private void applyModificationForAlignedChunkMetadataList(
       return;
     }
     IDeviceID device = currentDevice.getLeft();
-    Deletion ttlDeletion = null;
+    ModEntry ttlDeletion = null;
     if (tsFileResource.getStartTime(device) < timeLowerBoundForCurrentDevice) {
       ttlDeletion =
-          new Deletion(
+          new TreeDeletionEntry(
```

Comment:
Should distinguish between tree models and table models for current device.

Reply:
Done